File: packages/js-sdk/examples/with-openai.ts
/**
 * OpenAI Integration Example
 *
 * This example shows how to use Blindfold with OpenAI to:
 * 1. Tokenize sensitive data before sending to OpenAI
 * 2. Get the response from OpenAI
 * 3. Detokenize the response to restore original data
 */

import { Blindfold } from '../src/index'
// Uncomment if you have OpenAI installed:
// import OpenAI from 'openai'

async function main() {
  // Initialize clients
  const blindfold = new Blindfold({
    apiKey: process.env.BLINDFOLD_API_KEY || 'your-blindfold-key',
  })

  // Uncomment if using OpenAI:
  // const openai = new OpenAI({
  //   apiKey: process.env.OPENAI_API_KEY,
  // })

  // User message with sensitive data
  const userMessage = `
    I need help with my account. My name is Sarah Johnson,
    my email is sarah.johnson@company.com, and my phone is +1-555-0199.
    I was born on 05/15/1990.
  `

  console.log('Original message:', userMessage)
  console.log()

  // Step 1: Tokenize sensitive data
  console.log('Step 1: Tokenizing sensitive data...')
  const tokenized = await blindfold.tokenize(userMessage, {
    policy: 'gdpr_eu', // Use GDPR policy
  })

  console.log('Tokenized message:', tokenized.text)
  console.log('Detected entities:')
  tokenized.detected_entities.forEach((entity) => {
    console.log(`  - ${entity.entity_type}: ${entity.text} (score: ${entity.score})`)
  })
  console.log()

  // Step 2: Send to OpenAI (simulated)
  console.log('Step 2: Sending to OpenAI...')

  // Uncomment to use real OpenAI:
  // const completion = await openai.chat.completions.create({
  //   model: 'gpt-4',
  //   messages: [
  //     {
  //       role: 'system',
  //       content: 'You are a helpful customer service assistant.',
  //     },
  //     {
  //       role: 'user',
  //       content: tokenized.text,
  //     },
  //   ],
  // })
  // const aiResponse = completion.choices[0].message.content || ''

  // Simulated response for demo
  const aiResponse = `Hello ${tokenized.text.match(/<Person_\d+>/)?.[0]}! I'd be happy to help you with your account. I'll send a verification email to ${tokenized.text.match(/<Email Address_\d+>/)?.[0]} and a text to ${tokenized.text.match(/<Phone Number_\d+>/)?.[0]}. For security, I'll also verify your date of birth ${tokenized.text.match(/<Date Of Birth_\d+>/)?.[0]}.`

  console.log('AI Response (tokenized):', aiResponse)
  console.log()

  // Step 3: Detokenize the response (client-side)
  console.log('Step 3: Detokenizing response...')
  const detokenized = blindfold.detokenize(aiResponse, tokenized.mapping)

  console.log('Final response:', detokenized.text)
  console.log()

  console.log('✅ Sensitive data was protected throughout the entire interaction!')
}

main().catch(console.error)

File: packages/python-sdk/examples/with_openai.py
"""
OpenAI Integration Example

This example shows how to use Blindfold with OpenAI to:
1. Tokenize sensitive data before sending to OpenAI
2. Get the response from OpenAI
3. Detokenize the response to restore original data
"""

import os
import re
import sys
from pathlib import Path

# Add parent directory to path to import from local source
sys.path.insert(0, str(Path(__file__).parent.parent / "src"))

from blindfold import Blindfold

# Uncomment if you have OpenAI installed:
# from openai import OpenAI


def main():
    # Initialize clients
    blindfold = Blindfold(
        api_key=os.getenv("BLINDFOLD_API_KEY", "your-blindfold-key")
    )

    # Uncomment if using OpenAI:
    # openai_client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))

    # User message with sensitive data
    user_message = """
    I need help with my account. My name is Sarah Johnson,
    my email is sarah.johnson@company.com, and my phone is +1-555-0199.
    I was born on 05/15/1990.
    """

    print(f"Original message: {user_message}")
    print()

    # Step 1: Tokenize sensitive data
    print("Step 1: Tokenizing sensitive data...")
    tokenized = blindfold.tokenize(text=user_message, policy="gdpr_eu")

    print(f"Tokenized message: {tokenized.text}")
    print("Detected entities:")
    for entity in tokenized.detected_entities:
        print(f"  - {entity.type}: {entity.text} (score: {entity.score:.2f})")
    print()

    # Step 2: Send to OpenAI (simulated)
    print("Step 2: Sending to OpenAI...")

    # Uncomment to use real OpenAI:
    # completion = openai_client.chat.completions.create(
    #     model="gpt-4",
    #     messages=[
    #         {
    #             "role": "system",
    #             "content": "You are a helpful customer service assistant.",
    #         },
    #         {"role": "user", "content": tokenized.text},
    #     ],
    # )
    # ai_response = completion.choices[0].message.content or ""

    # Simulated response for demo
    person_token = re.search(r"<Person_\d+>", tokenized.text)
    email_token = re.search(r"<Email Address_\d+>", tokenized.text)
    phone_token = re.search(r"<Phone Number_\d+>", tokenized.text)
    date_token = re.search(r"<Date Of Birth_\d+>", tokenized.text)

    ai_response = f"Hello {person_token.group() if person_token else ''}! I'd be happy to help you with your account. I'll send a verification email to {email_token.group() if email_token else ''} and a text to {phone_token.group() if phone_token else ''}. For security, I'll also verify your date of birth {date_token.group() if date_token else ''}."

    print(f"AI Response (tokenized): {ai_response}")
    print()

    # Step 3: Detokenize the response (client-side)
    print("Step 3: Detokenizing response...")
    detokenized = blindfold.detokenize(ai_response, tokenized.mapping)

    print(f"Final response: {detokenized.text}")
    print()

    print("✅ Sensitive data was protected throughout the entire interaction!")


if __name__ == "__main__":
    main()
